tests/js/common/aql/aql-view-arangosearch-feature.js

```diff
@@ -55,42 +55,99 @@ function iResearchFeatureAqlTestSuite () {
         let analyzer = analyzers.save("stemPropAnalyzer", "stem", { "locale":"en", "invalid_param":true});
         assertTrue(null != analyzer);
         assertTrue(null == analyzer.properties.invalid_param);
-        analyzers.remove(db._name() + "::stemPropAnalyzer", true);
+        analyzers.remove("stemPropAnalyzer", true);
       }
       {
-        try {analyzers.remove(db._name() + "::ngramPropAnalyzer"); } catch (e) {}
+        try {analyzers.remove("ngramPropAnalyzer"); } catch (e) {}
         let analyzer = analyzers.save("ngramPropAnalyzer", "ngram", { "min":1, "max":5, "preserveOriginal":true, "invalid_param":true});
         assertTrue(null != analyzer);
         assertTrue(null == analyzer.properties.invalid_param);
-        analyzers.remove(db._name() + "::ngramPropAnalyzer", true);
+        analyzers.remove("ngramPropAnalyzer", true);
       }
     },
-
+    testAnalyzerRemovalWithDatabaseName_InSystem: function() {
+      let dbName = "analyzerWrongDbName";
+      db._useDatabase("_system");
+      try { db._dropDatabase(dbName); } catch (e) {}
+      db._createDatabase(dbName);
+      db._useDatabase(dbName);
+      analyzers.save("MyTrigram", "ngram", { min: 2, max: 3, preserveOriginal: true });
+      db._useDatabase("_system");
+      try {
+        analyzers.remove(dbName + "::MyTrigram");
+        fail(); // removal with db name in wrong current used db should also fail
+      } catch(e) {
+        assertEqual(require("internal").errors.ERROR_BAD_PARAMETER .code,
+                       e.errorNum);
+      }
+      db._dropDatabase(dbName);
+    },
+    testAnalyzerRemovalWithDatabaseName_InDbName: function() {
+      let dbName = "analyzerWrongDbName";
+      db._useDatabase("_system");
+      try { db._dropDatabase(dbName); } catch (e) {}
+      db._createDatabase(dbName);
+      db._useDatabase(dbName);
+      analyzers.save("MyTrigram", "ngram", { min: 2, max: 3, preserveOriginal: true });
+      try {
+        analyzers.remove(dbName + "::MyTrigram");
+        fail(); // removal with db name should  fail
+      } catch(e) {
+        assertEqual(require("internal").errors.ERROR_BAD_PARAMETER .code,
+                       e.errorNum);
+      }
+      db._useDatabase("_system");
+      db._dropDatabase(dbName);
+    },
+    testAnalyzerGetFromOtherDatabase: function() {
+      let dbName = "analyzerDbName";
+      let anotherDbName = "anotherDbName";
+      db._useDatabase("_system");
+      try { db._dropDatabase(dbName); } catch (e) {}
+      try { db._dropDatabase(anotherDbName); } catch (e) {}
+      db._createDatabase(dbName);
+      db._createDatabase(anotherDbName);
+      db._useDatabase(dbName);
+      let analyzer = analyzers.save("MyTrigram", "ngram", { min: 2, max: 3, preserveOriginal: true });
+      assertTrue(null != analyzer);
+      analyzer = undefined;
+      db._useDatabase(anotherDbName);
+      try {
+        analyzer = analyzers.analyzer(dbName + "::MyTrigram");
```

Comment:
no need to assign the analyzer to variable `analyzer` here, as the value is never used afterwards.